integration-test/mqtt/mqtt_system_test.c

```diff
@@ -392,15 +410,26 @@ static void establishMqttSession( MQTTContext_t * pContext,
 
     if( useLWTClientIdentifier )
     {
-        connectInfo.pClientIdentifier = TEST_CLIENT_IDENTIFIER_LWT;
-        connectInfo.clientIdentifierLength = TEST_CLIENT_IDENTIFIER_LWT_LENGTH;
+        int randNum = ( rand() % MAX_RAND_NUMBER_IN_FOR_CLIENT_ID );
+
+        /* Populate client identifier for connection with LWT topic with random number. */
+        connectInfo.clientIdentifierLength =
+            snprintf( clientIdBuffer, sizeof( clientIdBuffer ), "%d%s", randNum, TEST_CLIENT_IDENTIFIER_LWT );
+        connectInfo.pClientIdentifier = clientIdBuffer;
     }
     else
     {
-        connectInfo.pClientIdentifier = TEST_CLIENT_IDENTIFIER;
-        connectInfo.clientIdentifierLength = TEST_CLIENT_IDENTIFIER_LENGTH;
+        int randNum = ( rand() % MAX_RAND_NUMBER_IN_FOR_CLIENT_ID );
```

Comment:
Will this affect session resumption tests that rely resuming a non-clean session?

Reply:
Probably yes. How about we populate client ID at the beginning of the demo so that it is fixed throughout?

Reply:
Good point. Have moved the call to `rand` in the `setUp` function.